central/reports/scheduler/schedule.go

```diff
@@ -285,9 +287,13 @@ func (s *scheduler) sendReportResults(req *ReportRequest) error {
 	}
 	// If it is an empty report, do not send an attachment in the final notification email and the email body
 	// will indicate that no vulns were found
-	messageText := noVulnsFoundEmailTemplate
+	messageText, err := formatMessage(rc, noVulnsFoundEmailTemplate)
+	if err != nil {
+		return err
+	}
+
 	if zippedCSVData != nil {
-		messageText, err = formatMessage(rc)
+		messageText, err = formatMessage(rc, vulnReportEmailTemplate)
 		if err != nil {
 			return errors.Wrap(err, "error formatting the report email text")
 		}
```

Comment:
Nitpick, but it seems the following would be more direct and produce less code duplication.

```go
templateStr := vulnReportEmailTemplate
if zippedCSVData == nil {
	// If it is an empty report, the email body will indicate that no vulns were found
	templateStr = noVulnsFoundEmailTemplate	
}

messageText, err := formatMessage(rc, templateStr)
if err != nil {
	return errors.Wrap(err, "error formatting the report email text")
}
```

Reply:
Fair, this also avoids a redundant call of `formatMessage`whenever `zippedCSVData` is not nil